python/lsst/afw/typehandling/_GenericMap.cc

```diff
@@ -74,7 +74,16 @@ template <typename K>
 py::object get(GenericMap<K>& self, K const& key) {
     auto callable = static_cast<typename Publicist<K>::ConstValueReference (GenericMap<K>::*)(K) const>(
             &Publicist<K>::unsafeLookup);
-    return py::cast((self.*callable)(key));
+    auto variant = (self.*callable)(key);
+    try {
+        // py::cast can't handle this implicit conversion; do it by hand
+        Storable const& value = boost::get<PolymorphicValue const&>(variant);
+        // Prevent segfaults when assigning a Key<Storable> to Python variable, then deleting from map
+        // No existing code depends on being able to modify an item stored by value
+        return py::cast(value.cloneStorable());
+    } catch (boost::bad_get const&) {
+        return py::cast(variant);
+    }
```

Comment:
There is. As noted in the main `GenericMap` code, I did not use it in the belief that there's no `std::variant` equivalent, but it turns out [it exists under another name](https://en.cppreference.com/w/cpp/utility/variant/get_if).

Reply:
Hmm... there's a problem. While I should be able to switch the the pointer-based version of `get` here and in the C++ accessor for `Storable`, if I do it in the `shared_ptr<Storable>` accessor as well, I can't easily distinguish between a variant that contains something other than a pointer and a variant that contains `nullptr` -- `get` will return null in both cases.

I could keep the throwing `get` there, or I could use [`which`](https://www.boost.org/doc/libs/1_71_0/doc/html/boost/variant.html#id-1_3_46_5_4_1_1_16_1-bb) (which requires an integer-based index, and is therefore brittle), or I could use [`type`](https://www.boost.org/doc/libs/1_71_0/doc/html/boost/variant.html#id-1_3_46_5_4_1_1_16_3-bb) (which involves more RTTI). I assume that will also be too much of a performance penalty for you?

Reply:
Would a direct `py::cast` (as in the `catch` block here) also do what you want in the case that the variant contains a `nullptr`?  After all, you can't clone that either.

If not, just leave as-is with the throwing `get`.

Reply:
I ended up only changing the pybind11 code. There's less incentive to use a pointer `get` in the C++ accessors (because the response on a mismatch is to throw an exception anyway), and the inability to implicitly convert `PolymorphicValue*` to `Storable*` makes the pointer-based code much more awkward. Somehow that proxy class isn't working out as well as one would expect... :disappointed: